File: src/main/kotlin/xyz/mahmoudahmed/dsl/BranchConditionBuilder.java
package xyz.mahmoudahmed.dsl;

import java.util.function.Predicate;

public class BranchConditionBuilder {
    private final BranchBuilder parent;
    private final Predicate<Object> condition;

    BranchConditionBuilder(BranchBuilder parent, Predicate<Object> condition) {
        this.parent = parent;
        this.condition = condition;
    }

    public BranchActionBuilder then() {
        return new BranchActionBuilder(parent, condition);
    }
}

File: src/main/kotlin/xyz/mahmoudahmed/dsl/conditional/BranchConditionBuilder.kt
package xyz.mahmoudahmed.dsl.conditional

/**
 * Builder for configuring a condition in a branch.
 */
class BranchConditionBuilder(
    private val parent: BranchBuilder,
    private val condition: (Any) -> Boolean
) {
    /**
     * Specifies the actions to perform when the condition is true.
     *
     * @return Builder for configuring branch actions
     */
    fun then(): BranchActionBuilder {
        return BranchActionBuilder(parent, condition)
    }
}